ui.frontend/src/main/webpack/site/js/components/searchInput.ts

```diff
@@ -45,19 +47,39 @@ const debouncedSearch = (autoSuggestionDebounceTime: number) =>
       searchInput: HTMLInputElement,
       searchContainer: HTMLDivElement,
     ) => {
-      removeSuggestionList(searchContainer)
       setSaasCurrentFocusSuggestion(searchInput, -1)
 
+      if (!query?.length || query.length < autocompleteTriggerThreshold) {
+        removeSuggestionList(searchContainer)
+      }
+
       if (query.length >= autocompleteTriggerThreshold) {
+        const regexp = new RegExp(query, 'gi')
+        const searchButtonElement = document.getElementsByClassName(
+          'saas-container_button',
+        )?.[0] as HTMLElement
         const results = await fetchAutoComplete(autocompleteUrl, query)
+        let suggestionDropdown: any = null
```

Comment:
isn't the type a `HTMLElement` ? Or div element?

Reply:
yes and it complains with null :D

Maybe I can do HTMLElement | null

Reply:
That should work 👍